### PR TITLE
CLI for Graph

### PR DESCRIPTION
**Pull request by BilgeDenizKocakk**

### CODE REVIEW

File: packages/lage/src/config/getConfig.ts [hunk 1]
```diff
@@ -76,6 +76,9 @@ export function getConfig(cwd: string): Config {
     continue: parsedArgs.continue || configResults?.config.continue,
     safeExit: parsedArgs.safeExit,
     includeDependencies: parsedArgs.includeDependencies,
+    includeDependents: parsedArgs.includeDependents,
+    withDevDependencies: parsedArgs.withDevDependencies,
+    withPeerDependencies: parsedArgs.withPeerDependenies,
     clear: parsedArgs.clear || false,
```
Comment (kerrynf) [inline packages/lage/src/config/getConfig.ts:82]: Default value question. :) When lage runs without any flags, what do each of these default to? (My guess is false?)

File: packages/lage/src/command/graph.ts [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
 
-export async function graph(cwd: string) {
+export async function graph(cwd: string, config: Config) {
+  const fs = require("fs");
   const root = getWorkspaceRoot(cwd);
   if (!root) {
     throw new Error("This must be called inside a codebase that is part of a JavaScript workspace.");
   }
   const allPackages = getPackageInfos(root);
+  const pattern = getConfig(cwd).scope;
```
Comment (kerrynf) [inline packages/lage/src/command/graph.ts:12]: If scope isn't set, what happens? Should we supply the other flags if the pattern is empty?

File: packages/lage/src/config/getConfig.ts (hunk 1 above)
Comment (BilgeDenizKocakk) [reply packages/lage/src/config/getConfig.ts:82]: The default value for a boolean in Typescript would be false, right? This is how PackageGraphFilter is defined in createPackageGraph within workspace-tools:

export interface PackageGraphFilter {
  namePatterns: string[];
  includeDependencies?: boolean;
  includeDependents?: boolean;
  withDevDependencies?: boolean;
  withPeerDependencies?: boolean;
}

File: packages/lage/src/command/graph.ts (hunk 2 above)
Comment (BilgeDenizKocakk) [reply packages/lage/src/command/graph.ts:12]: I tested this within createPackageGraph and it was returning the entire graph, however, the command returns an empty graph -- plus it looks like there are errors in getting the development and peer dependencies using the command flags. I will be looking into this in more detail!

File: packages/lage/src/config/getConfig.ts (hunk 1 above)
Comment (kenotron) [reply packages/lage/src/config/getConfig.ts:82]: For actually running the commands, e.g. `lage build --scope pkg-a`:

devdeps should default to true, 
while the peerdeps mean very little in this context - and should be false (peer deps are usually things that are provided by something external, but I suppose SOMETIMES you are lucky that your dependencies are telling you to dep on something that already exists in your monorepo)

Comment (kenotron) [reply packages/lage/src/config/getConfig.ts:82]: In this case, defaulting devDeps to true please:

```ts
withDevDependencies: parsedArgs.withDevDependencies !== false
```

File: packages/lage/src/command/graph.ts (hunk 2 above)
Comment (kenotron) [reply packages/lage/src/command/graph.ts:12]: I think getConfig might fill out scope to something by default?

Comment (BilgeDenizKocakk) [reply packages/lage/src/command/graph.ts:17]: I just added an if else statement that checks if the pattern is empty/undefined and sets 
* the namePatterns as all packages
* includeDependencies, includeDependents, withDevDependencies, and withPeerDependencies to true 
so if there is no scope provided, it shows the entire graph!

**Comment by kenotron** [comment]
Closing this for now - we'll add this back when we have a chance to properly support it

**Pull request closed without merging**